bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerCreateOp.java

```diff
@@ -200,4 +209,233 @@ private void createComplete(int rc, LedgerHandle lh) {
+        }
+
+        @Override
+        public CreateBuilder withAckQuorumSize(int ackQuorumSize) {
+            this.builderAckQuorumSize = ackQuorumSize;
+            return this;
+        }
+
+        @Override
+        public CreateBuilder withPassword(byte[] password) {
+            this.builderPassword = password;
+            return this;
+        }
+
+        @Override
+        public CreateBuilder withCustomMetadata(Map<String, byte[]> customMetadata) {
+            this.builderCustomMetadata = customMetadata;
+            return this;
+        }
+
+        @Override
+        public CreateBuilder withDigestType(org.apache.bookkeeper.client.api.DigestType digestType) {
+            this.builderDigestType = digestType;
+            return this;
+        }
+
+        @Override
+        public CreateAdvBuilder makeAdv() {
+            if(advBuilder == null) {
+                advBuilder = new CreateAdvBuilderImpl(this);
+            }
+            return advBuilder;
+        }
+
+        private boolean validate(CreateCallback cb, Object ctx) {
+            if (builderWriteQuorumSize > builderEnsembleSize) {
+                LOG.error("invalid writeQuorumSize {} > ensembleSize {}", builderWriteQuorumSize, builderEnsembleSize);
+                cb.createComplete(BKException.Code.IncorrectParameterException, null, ctx);
+                return false;
+            }
+
+            if (builderAckQuorumSize > builderWriteQuorumSize) {
+                LOG.error("invalid ackQuorumSize {} > writeQuorumSize {}", builderAckQuorumSize, builderWriteQuorumSize);
+                cb.createComplete(BKException.Code.IncorrectParameterException, null, ctx);
+                return false;
+            }
+
+            if (builderAckQuorumSize <= 0) {
+                LOG.error("invalid ackQuorumSize {} <= 0", builderAckQuorumSize);
+                cb.createComplete(BKException.Code.IncorrectParameterException, null, ctx);
+                return false;
+            }
+
+            if (builderPassword == null) {
+                LOG.error("invalid null password");
+                cb.createComplete(BKException.Code.IncorrectParameterException, null, ctx);
+                return false;
+            }
+
+            if (builderDigestType == null) {
+                LOG.error("invalid null digestType");
+                cb.createComplete(BKException.Code.IncorrectParameterException, null, ctx);
+                return false;
+            }
+
+            if (builderCustomMetadata == null) {
+                LOG.error("invalid null customMetadata");
+                cb.createComplete(BKException.Code.IncorrectParameterException, null, ctx);
+                return false;
+            }
+
+            return true;
+        }
+
+        @Override
+        public CompletableFuture<WriteHandle> execute() {
+            CompletableFuture<WriteHandle> counter = new CompletableFuture<>();
+            create(new SyncCreateCallback(), counter);
```

Comment:
I think I suggest avoid using ctx object for the completablefuture. you didn't address my comment.

also counter is a wrong name.

Reply:
done